File: java/idades.java
import java.util.Locale;
import java.util.Scanner;

public class idades {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		String nome1, nome2;
		int idade1, idade2;
		double media;

		System.out.println("Dados da primeira pessoa:");
		System.out.print("Nome: ");
		nome1 = sc.nextLine();
		System.out.print("Idade: ");
		idade1 = sc.nextInt();

		System.out.println("Dados da segunda pessoa:");
		System.out.print("Nome: ");
		sc.nextLine(); // --------------> limpeza de buffer
		nome2 = sc.nextLine();
		System.out.print("Idade: ");
		idade2 = sc.nextInt();

		media = (idade1 + idade2) / 2.0;

		System.out.printf("A idade media de %s e %s eh de %.1f anos", nome1, nome2, media);

		sc.close();
	}
}

File: python/idades.py
nome1: str ; nome2: str
idade1: int ; idade2: int
media: float

print("Dados da primeira pessoa:")
nome1 = str(input("Nome: "))
idade1 = int(input("Idade: "))

print("Dados da segunda pessoa:")
nome2 = str(input("Nome: "))
idade2 = int(input("Idade: "))

media = (idade1 + idade2) / 2.0

print(f"A idade media de {nome1} e {nome2} eh de {media:.1f} anos")